### DISCUSSION TITLE
[Request] Allow for unmanaged constructor arguments

### DISCUSSION THREAD

**Discussion by mscharley** [mscharley/dot#54] [Ideas]
### Is your feature request related to a problem? Please describe it.

Sometimes it is useful to have constructor parameters which have a default value for the sake of mocking out libraries.

### Describe the solution you'd like.

```typescript
import library from "some-library";
import { unmanaged } from "@mscharley/ioc";

@injectable(unmanaged)
class Test {
  public constructor(private lib = library) {
    this.lib.doSomething();
  }
}
```

### Describe alternatives you've considered.

You could force everything to go through a token, but for injecting a whole library this feels strange and would rely on either:

1. `import('some-library')` types which can be strange, especially in this transitional period between CJS and ESM.
2. Redefine the library as an interface which the application then has to define and maintain which feels even worse.

**Comment by VasyaVoronn** [comment]
Hi i want to help you

**Reply by mscharley** [reply]
Hey! For now, the best way to help is probably to actually use the project. As of right now, I don't have any real projects using this library yet, but it has been built with what I need in order to migrate some existing projects over to using it in the future. But at the moment, to my knowledge, no one is using it in production yet and that is what will help find the gaps that need to be filled.

If you wanted to help out with this idea in particular then go for it, PR's are open and I haven't started work on this at all yet.

The other big place that people can help out is documentation, there's several large holes in the API documentation right now.

**Comment by mscharley** [comment]
Implemented in #69
